File: 1072/Solution.go
func maxEqualRowsAfterFlips(matrix [][]int) int {
    hm := make(map[string]int)

    for _, r := range matrix {
        key := make([]byte, len(r))
        for i, n := range r{
            if r[0] == 1{
                key[i] = byte('0' + (1 - n))
            }else{
                key[i] = byte('0' + n)
            }
        }
        hm[string(key)]++
    }

    res := 0
    for _, v := range hm {
        res = max(res, v)
    }
    return res
}

File: 1072/Solution.java
class Solution {
    public int maxEqualRowsAfterFlips(int[][] matrix) {
        Map<String, Integer> hm = new HashMap<>();
        for(int[] r : matrix) {
            StringBuilder sb = new StringBuilder();
            if(r[0]==1){
                for(int i : r){
                    sb.append( i == 0 ? 1 : 0); 
                }
            }else{
                for(int i : r) sb.append(i);
            }
            hm.merge(sb.toString(), 1, Integer::sum);
        }

        return Collections.max(hm.values());

    }
}
